src/js/index.js

```diff
@@ -1,14 +1,195 @@
+    createNotFoundDiv,
+    createSearchKeyWordElement
+} from "./view.js";
+import {
+    deleteStoredItem,
+    getFetchedItems,
+    toggleWatchStoredItem,
+    resetFetchedItems,
+    remainSearchedKeyword3,
+    getNextPageToken,
+    addFetchedItems,
+    saveStoredItem,
+    toggleLikeStoredItem,
+    getSearchedKeyword,
+    getStoredItems
+} from "./model.js";
+
+// config
+let apiKey = key;
+let q = "";
+let requestURL = ``;
+
+
+// view
 const $searchButton = document.querySelector("#search-button");
 const $modalClose = document.querySelector(".modal-close");
 const $modal = document.querySelector(".modal");
+const $fetchButton = document.querySelector("#fetch-button");
+const $fetchInput = document.querySelector("#fetch-input");
+const $modalInner = document.querySelector(".modal-inner");
+const $fetchedContainer = document.querySelector("#fetched-container");
+const $storedContainer = document.querySelector("#stored-container");
+
+const $mainPage = document.querySelector('#mainPage');
+const $watchedPage = document.querySelector('#watchedPage');
+const $likedPage = document.querySelector('#likedPage');
+
 
+
+// controller
 const onModalShow = () => {
-  $modal.classList.add("open");
+    $modal.classList.add("open");
+    createSearchKeyWordElement(getSearchedKeyword());
 };
 
 const onModalClose = () => {
-  $modal.classList.remove("open");
+    $modal.classList.remove("open");
 };
 
+const onFetchItemList = (e) => {
+    e.preventDefault();
+    q = $fetchInput.value;
+    remainSearchedKeyword3(q);
+    createSearchKeyWordElement(getSearchedKeyword());
+    requestURL = `https://www.googleapis.com/youtube/v3/search?key=${apiKey}&part=snippet&maxResults=10&q=${q}`; //
+    $fetchInput.value = "";
+
+    fetch(requestURL).then(res => res.json()).then(data => {
+        resetFetchedItems(data); // model call
+
+        if (getFetchedItems().length === 0) {
+            createNotFoundDiv(); // view call
+        }
+        createModalArticles(getFetchedItems(), getStoredItems()); // view call
+    });
+}
+
+
+const onFetchItemListWithEnter = (e) => {
+    if (e.keyCode == 13) {
+        e.preventDefault();
+        $fetchButton.click();
+    }
+}
+let throttle;
+const onFetchNextPageItemList = () => {
+    if (parseInt($modalInner.scrollTop + $modalInner.clientHeight + 10) > parseInt($modalInner.scrollHeight)) {
+        if (!throttle) {
+            console.log("페칭");
+            throttle = setTimeout(() => {
+                throttle = null;
+                requestURL = `https://www.googleapis.com/youtube/v3/search?key=${apiKey}&part=snippet&maxResults=10&q=${q}&pageToken=${getNextPageToken()}`;
+                fetch(requestURL).then(res => res.json()).then(data => {
+                    addFetchedItems(data);
+                    createModalArticles(getFetchedItems(), getStoredItems());
+                    getFetchedItems().nextPageToken = data.nextPageToken;
+                })
+            }, 1000);
+        }
+    }
```

Comment:
```suggestion
    if (parseInt($modalInner.scrollTop + $modalInner.clientHeight + 10) <= parseInt($modalInner.scrollHeight)) return;
    if (throttle) return;
    console.log("페칭");
    throttle = setTimeout(() => {
        throttle = null;
        requestURL = `https://www.googleapis.com/youtube/v3/search?key=${apiKey}&part=snippet&maxResults=10&q=${q}&pageToken=${getNextPageToken()}`;
        fetch(requestURL).then(res => res.json()).then(data => {
            addFetchedItems(data);
            createModalArticles(getFetchedItems(), getStoredItems());
            getFetchedItems().nextPageToken = data.nextPageToken;
        })
    }, 1000);
```
이렇게 if로 인하여 코드의 depth가 깊어지지 않도록 작성해주세요! 그래야 가독성을 좋게 유지할 수있 습니다.